agent-control/src/cli/on_host/config_gen/config.rs

```diff
@@ -36,6 +34,46 @@ impl From<AgentSet> for HashMap<String, Agent> {
     }
 }
 
+/// Holds the proxy configuration.
+/// Cannot use [crate::http::config::ProxyConfig] directly due lack of support for defaults in clap.
+/// See <https://github.com/clap-rs/clap/issues/4746> for details.
+#[derive(Debug, Default, Clone, PartialEq, Serialize, clap::Args)]
+pub struct ProxyConfig {
+    #[serde(skip_serializing_if = "is_none_or_empty_string", rename = "url")]
+    #[arg(long, required = false)]
+    pub proxy_url: Option<String>,
+
+    #[serde(
+        skip_serializing_if = "is_none_or_empty_string",
+        rename = "ca_bundle_dir"
+    )]
+    #[arg(long, required = false)]
+    pub proxy_ca_bundle_dir: Option<String>,
+
+    #[serde(
+        skip_serializing_if = "is_none_or_empty_string",
+        rename = "ca_bundle_file"
+    )]
+    #[arg(long, required = false)]
+    pub proxy_ca_bundle_file: Option<String>,
+
+    #[arg(long, default_value_t = false, value_parser = ignore_system_proxy_parser, action = clap::ArgAction::Set)]
+    pub ignore_system_proxy: bool,
+}
+
+// Helper to avoid serializing empty values
+fn is_none_or_empty_string(v: &Option<String>) -> bool {
+    v.as_ref().map(|s| s.is_empty()).unwrap_or(true)
+}
+
+// Custom parser to allow empty values as false booleans
+fn ignore_system_proxy_parser(s: &str) -> Result<bool, Infallible> {
+    match s.to_lowercase().as_str() {
+        "" | "false" => Ok(false),
+        _ => Ok(true),
+    }
+}
```

Comment:
👌 using `Infallible`.

I guess this is because `value_parser` is required to return a `Result`?